src/utils.ts

```diff
@@ -82,65 +76,47 @@ export function isValidCode(code: number): boolean {
  */
 export function serializeError(
   error: unknown,
-  { fallbackError = FALLBACK_ERROR, shouldIncludeStack = false } = {},
-): SerializedEthereumRpcError {
-  if (
-    !fallbackError ||
-    !Number.isInteger(fallbackError.code) ||
-    typeof fallbackError.message !== 'string'
-  ) {
+  { fallbackError = FALLBACK_ERROR, shouldIncludeStack = true } = {},
+): JsonRpcError {
+  if (!isJsonRpcError(fallbackError)) {
     throw new Error(
       'Must provide fallback error with integer number code and string message.',
     );
   }
 
-  if (error instanceof EthereumRpcError) {
-    return error.serialize();
-  }
+  const serialized = buildError(error, fallbackError as JsonRpcError);
```

Comment:
Nit: This cast does not seem necessary
```suggestion
  const serialized = buildError(error, fallbackError);
```